components/authz-service/engine/opa/policy/authz_v2.rego

```diff
@@ -102,14 +92,7 @@ denied_project[project] {
 	has_project[[project, pol_id, statement_id]]
 }
 
-# when no input is provided, this rule short-circuits checking any other allowed projects
-denied_all_projects {
-	match[["deny", pol_id, statement_id]]
-	common.const_all_projects == policies[pol_id].statements[statement_id].projects[_]
-}
-
 authorized_project[project] {
-	not denied_all_projects
```

Comment:
I was going to say, "Wait, wait; this is needed for a special case!"  But not anymore. 😁